samples/tutorials/mnist/README.md

```diff
@@ -0,0 +1,110 @@
+    - Use the Kubeflow Pipelines SDK to load your component, use it in a pipeline and run that pipeline.
+
+- [04 Reusable and Pre-build Components as Pipeline](04_Reusable_and_Pre-build_Components_as_Pipeline.ipynb): this 
+notebook combines our built components, together with a pre-build GCP AI Platform components 
+and a lightweight component to compose a pipeline with three steps.
+    - Train a MNIST model and export it to GCS
+    - Deploy the exported Tensorflow model on AI Platform prediction service
+    - Test the deployment by calling the end point with test data
+
+## Setups Overview:
+### Prerequisites
+Before you follow the instructions below to deploy your own Kubeflow cluster, you should
+
+- have a [GCP project setup](https://www.kubeflow.org/docs/gke/deploy/project-setup/) for your Kubeflow deployment 
+with you having the [owner role](https://cloud.google.com/iam/docs/understanding-roles#primitive_role_definitions) 
+for the project and with the following APIs enabled:
+    - [Compute Engine API](https://pantheon.corp.google.com/apis/library/compute.googleapis.com)
+    - [Kubernetes Engine API](https://pantheon.corp.google.com/apis/library/container.googleapis.com)
+    - [Identity and Access Management(IAM) API](https://pantheon.corp.google.com/apis/library/iam.googleapis.com)
+    - [Deployment Manager API](https://pantheon.corp.google.com/apis/library/deploymentmanager.googleapis.com)
+    - [Cloud Resource Manager API](https://pantheon.corp.google.com/apis/library/cloudresourcemanager.googleapis.com)
+    - [Cloud Filestore API](https://pantheon.corp.google.com/apis/library/file.googleapis.com)
+    - [AI Platform Training & Prediction API](https://pantheon.corp.google.com/apis/library/ml.googleapis.com)
+- have set up [OAuth for Cloud IAP](https://www.kubeflow.org/docs/gke/deploy/oauth-setup/)
+- have installed and setup [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- have installed [gcloud-sdk](https://cloud.google.com/sdk/)
+
+### Setup Environment and Deploy Kubeflow Cluster
+#### Setup Environment
+* Download kfctl
+* Create user credentials
+* Setup environment variables
+* Create dedicated service account for deployment
+```bash
+export SA_NAME = [service account name]
+gcloud iam service-accounts create ${SA_NAME}
+gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member serviceAccount:${SA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com \
+    --role 'roles/owner'
+gcloud iam service-accounts keys create ~/key.json \
+    --iam-account ${SA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com
+export GOOGLE_APPLICATION_CREDENTIALS=~/key.json
+```
+
+#### Deploy Kubeflow Cluster
+* Deploy kubeflow
+```bash
+mkdir -p ${KF_DIR}
+cd $kf_dir
+kfctl apply -V -f ${CONFIG_URI}
+```
+
+* Install the lastest version of kfp
+```bash
+pip3 install kfp --upgrade --user
+```
+
+## Running Notebook
```

Comment:
Hi @joeliedtke , similar to the previous section, the intention of this section is also to help readers understand whether their existing (or preferred but not deployed yet) setup can be easily adopted, before going into the content of the notebook. I feel it would be better to stay in Readme instead of the notebooks, or we can duplicate them in the notebook if you think they should be there also.

In terms of recommendation, I personally feel that it's better to not recommend and let the user to choose based on their preference. Hope this makes sense.